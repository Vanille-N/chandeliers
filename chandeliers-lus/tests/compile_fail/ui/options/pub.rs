//! Not enough visibility on `PRIVATE` and `private`.
//! Need `pub` for them to escape their module.
#![allow(unused_imports)]

mod vis {
    chandeliers_lus::decl! {
        #[export]
        node private() returns ();
        let tel;

        #[export]
        const PRIVATE: int = 0;

        #[pub]
        node public() returns ();
        let tel;

        #[pub]
        const PUBLIC : int = 0;
    }
}

use vis::private;
use vis::PRIVATE;

use vis::public;
use vis::PUBLIC;

fn main() {}
