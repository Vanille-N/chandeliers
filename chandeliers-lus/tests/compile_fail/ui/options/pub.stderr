error[E0603]: struct `private` is private
  --> tests/compile_fail/ui/options/pub.rs:23:10
   |
23 | use vis::private;
   |          ^^^^^^^ private struct
   |
note: the struct `private` is defined here
  --> tests/compile_fail/ui/options/pub.rs:8:14
   |
8  |         node private() returns ();
   |              ^^^^^^^

error[E0603]: constant `PRIVATE` is private
  --> tests/compile_fail/ui/options/pub.rs:24:10
   |
24 | use vis::PRIVATE;
   |          ^^^^^^^ private constant
   |
note: the constant `PRIVATE` is defined here
  --> tests/compile_fail/ui/options/pub.rs:12:15
   |
12 |         const PRIVATE: int = 0;
   |               ^^^^^^^
